### PR TITLE
Make the freshness lifetime of an http response <= dns ttl

### PR DESCRIPTION
**Pull request by mcmanus**
closes issue #13

### CODE REVIEW

File: draft-ietf-doh-dns-over-https-latest.mkd [hunk 1]
```diff
@@ -288,10 +289,22 @@ responses. Responses may be returned in a different temporal order than
 requests were made using the protocols native multi-streaming
 functionality.
 
-In the HTTP responses, the HTTP cache headers SHOULD be set to expire
-at the same time as the shortest DNS TTL in the response. Because DNS
-provides only caching but not revalidation semantics, DNS over HTTP
-responses should not carry revalidation response headers
+A DNS response may contain a set of DNS Resource Records each with its
+own Time To Live (TTL) freshness information. The freshness lifetime
+({{RFC7234} Section 4.2) of the HTTP response should be coordinated
```
Comment (chantra) [inline draft-ietf-doh-dns-over-https-latest.mkd]: missing `}` in `{{RFC7234}}`

File: draft-ietf-doh-dns-over-https-latest.mkd [hunk 2]
```diff
@@ -288,10 +289,22 @@ responses. Responses may be returned in a different temporal order than
...
+with the Resource Record bearing the smallest TTL in the response. The
+freshness lifetime SHOULD be set to expire at the same time
+as the DNS Record has a 0 TTL. The response freshness lifetime MUST
```
Comment (chantra) [inline draft-ietf-doh-dns-over-https-latest.mkd]: s/as the DNS Record has a 0 TTL/as any of the DNS Records reach a TTL of 0/?

File: draft-ietf-doh-dns-over-https-latest.mkd [hunk 3]
```diff
@@ -288,10 +289,22 @@ responses. Responses may be returned in a different temporal order than
...
+NOT be greater than that indicated by the DNS Record with the smallest
+TTL in the response.
+
+A DNS API Client that receives a response without an explicit
+fresheness lifetime MUST NOT assign that response a heuristic
```
Comment (chantra) [inline draft-ietf-doh-dns-over-https-latest.mkd]: s/`fresheness`/`freshness`/

**Comment by mcmanus** [comment]
@chantra thanks.

**Pull request merged**